File: python/binary_search/FindPeakElement.py
class Solution:
    # @param num, a list of integer
    # @return an integer
    def findPeakElement(self, num):
        INT_MIN = -2147483648
        num.insert(0, INT_MIN)
        num.append(INT_MIN)
        
        low = 0
        high = len(num) - 1
        while low <= high:
            mid = (low + high) / 2
            if num[mid] > num[mid-1] and num[mid] > num[mid+1]:
                return mid - 1
            elif num[mid-1] > num[mid]:
                high = mid - 1
            else:
                low = mid + 1

if __name__ == "__main__":
    solution = Solution()
    A = [-2147483648]
    print solution.findPeakElement(A)

File: java/FindPeakElement.java
public class Solution {
    public int findPeakElement(int[] num) {
        int low = 0;
        int high = num.length-1;
        int mid = 0;
        while (low <= high) {
            mid = (low + high) / 2;
            if (mid != 0 && num[mid-1]>num[mid]) {
                high = mid - 1;
            } else if (mid != num.length-1 && num[mid+1] > num[mid]) {
                low = mid + 1;
            } else {
                return mid;
            }
        }
        return mid;
    }
}
